File: backend/app/models/notification.py
from pydantic import BaseModel, Field
from typing import Optional


class NotificationDB(BaseModel):
    """Notification record as stored in Supabase."""
    id: str
    user_id: str
    title: str
    body: str
    type: str = Field(..., description="One of: anomaly, credit_issued, system")
    related_id: Optional[str] = None
    is_read: bool = False
    created_at: Optional[str] = None


class NotificationCreate(BaseModel):
    """Internal model for creating a notification."""
    user_id: str
    title: str
    body: str
    type: str
    related_id: Optional[str] = None

File: android/app/src/main/java/com/carboncredit/app/data/models/Notification.kt
package com.carboncredit.app.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Notification(
    val id: String = "",
    @SerialName("user_id") val userId: String = "",
    val title: String = "",
    val body: String = "",
    val type: String = "",
    @SerialName("related_id") val relatedId: String? = null,
    @SerialName("is_read") val isRead: Boolean = false,
    @SerialName("created_at") val createdAt: String = ""
)
